src/aspire/operators/polar_ft.py

```diff
@@ -116,15 +99,46 @@ def _evaluate_t(self, x):
             Fourier grid. This is an array of vectors whose first dimension
             corresponds to `x.shape[0]`, and last dimension equals `self.count`.
         """
-        nimgs = x.shape[0]
-
-        half_size = self.ntheta // 2
-
-        pf = nufft(x, self.freqs)
+        if x.dtype != self.dtype:
+            raise TypeError(
+                f"{self.__class__.__name__}::transform"
+                f" Inconsistent dtypes x: {x.dtype} self: {self.dtype}"
+            )
+
+        if not isinstance(x, Image):
+            raise TypeError(
+                f"{self.__class__.__name__}::transform"
+                f" passed numpy array instead of {Image}."
+            )
+        else:
+            x = x.asnumpy()
+
+        # Flatten stack
+        stack_shape = x.shape[: -self.ndim]
+        x = x.reshape(-1, *x.shape[-self.ndim :])
+
+        # We expect the Image `x` to be real in order to take advantage of the conjugate
+        # symmetry of the Fourier transform of a real valued image.
+        if not np.isreal(x).all():
+            raise TypeError(
+                f"The Image `x` must be real valued. Found dtype {x.dtype}."
+            )
+
+        resolution = x.shape[-1]
+
+        pf = nufft(x, self.freqs) / resolution**2
+
+        return pf.reshape(*stack_shape, self.ntheta // 2, self.nrad)
+
+    @staticmethod
+    def full(pf):
```

Comment:
Would it be easier (simpler?) if this was just a kwarg to `transform`. I.e. calling `transform(full=True)` would give the full transform?

Reply:
I'm kind of split on this. We currently have algorithms that get the cross correlation between the half_transform and the full_transform. With this current implementation we would only compute the half_transform and then infer the full_transform using conjugate symmetry with `full(pf)` when needed. This just puts the computation for those uses in one (tested) location. 

If we make this a kwarg of transform we would be recomputing the full transform when calling `transform(full=True)`, which is something we should avoid if possible.  We could cache the half_transform and use it if it exists to compute the full_transform, but I think this begins to make things more complicated than necessary. 

What do you think?

Reply:
Well if the user wants both full and half, can't they get the full transform and split it?

Reply:
Sure, they could. The main user is currently our own algorithms. As the algorithms stand now, we mostly use the half and build the full when needed. 

I think ultimately we should just decide if we want `transform` to return the "full" transform or the "half". If it's full then we would store the larger data set and slice it when we need the half. If it's half we store less data and build the full when needed. 

Either way, for the sake of consistency we would probably still want a reusable method to get us from full-to-half or half-to-full (this would be true with or without a kwarg).

Reply:
Ok makes sense, but maybe rename it `half_to_full` so that it's clear that it takes an input and transforms it (since it's a method, it might otherwise look like it does `transform` but returns the full version).

Reply:
Ah good point. Will do.

Reply:
Done!

Reply:
:+1: